### DISCUSSION TITLE
Is it possible to just check out the source for a package?

### DISCUSSION THREAD

**Discussion by jalfd** [microsoft/vcpkg#42019] [Q&A]
As the title states, I would like to retrieve the source code, with all patches applied, for a given vcpkg package, and I'm not aware of any easy ways to do this.

The motivation is so that I can debug and step into code in that dependency and follow it in the debugger.

We use binary caching quite heavily because some of our dependencies (qtwebengine) takes hours to build, so I would like to be able to extract the source code even though I've already downloaded the pre-built binaries.

I could just manually check out the source and apply each and every patch listed in the portfile, it just seems like something that should be easier to do.

**Comment by dg0yt** [comment]
Did you try `--only-downloads`?
